source/game/field/ObjectDirector.cc

```diff
@@ -42,10 +48,70 @@ void ObjectDirector::addObject(ObjectCollidable *obj) {
     m_objects.push_back(obj);
 }
 
+/// @addr{0x8082AB04}
+u32 ObjectDirector::checkKartObjectCollision(Kart::KartObject *kartObj,
+        ObjectCollisionConvexHull *convexHull) {
+    u32 count = 0;
+    ObjectCollidable *obj;
+
+    while ((obj = reinterpret_cast<ObjectCollidable *>(
+                    BoxColManager::Instance()->getNextObject()))) {
```

Comment:
```suggestion
    while ((ObjectCollidable *obj = BoxColManager::Instance()->getNextObject())) {
```
We are guaranteeing that all objects are ObjectCollidable when we get them through this iterator. Better not to reinterpret_cast here, and instead reinterpret_cast in the return value.

Reply:
My compiler yells at me if I declare inside the while condition